eea/src/main/java/org/web3j/protocol/eea/crypto/PrivateTransactionEncoder.java

```diff
@@ -61,35 +59,25 @@ public static byte[] encode(final RawPrivateTransaction rawTransaction, final lo
     }
 
     private static byte[] encode(
-            final RawPrivateTransaction rawTransaction, final Sign.SignatureData signatureData) {
-        final List<RlpType> values = asRlpValues(rawTransaction, signatureData);
+            final RawPrivateTransaction privateTransaction,
+            final Sign.SignatureData signatureData) {
+        //        final List<RlpType> values = asRlpValues(rawTransaction, signatureData);
+        final List<RlpType> values = privateTransaction.asRlpValues(signatureData);
         final RlpList rlpList = new RlpList(values);
-        return RlpEncoder.encode(rlpList);
+        byte[] encoded = RlpEncoder.encode(rlpList);
+
+        if (privateTransaction.getTransactionType().isPrivateEip1559()) {
+            return ByteBuffer.allocate(encoded.length + 1)
+                    .put(privateTransaction.getTransactionType().getRlpType())
+                    .put(encoded)
+                    .array();
+        }
+        return encoded;
     }
 
     private static byte[] longToBytes(long x) {
         ByteBuffer buffer = ByteBuffer.allocate(Long.BYTES);
         buffer.putLong(x);
         return buffer.array();
     }
-
-    public static List<RlpType> asRlpValues(
```

Comment:
Any issue with this being removed for backwards compatibility?

Reply:
So every Tx type has its own different asRlpValues(), so it is being transferred from here.